tests/components/niko_home_control/test_init.py

```diff
@@ -0,0 +1,27 @@
+"""Test init."""
+
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.niko_home_control.const import DOMAIN
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+async def test_migrate_entry(
+    hass: HomeAssistant, entity_registry: er.EntityRegistry
+) -> None:
+    """Validate that the unique_id is migrated to the new unique_id."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        minor_version=1,
+        unique_id="light_1",
+        data={CONF_HOST: "192.168.0.123"},
+    )
+    config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    entity_entry = entity_registry.async_get_or_create(
+        LIGHT_DOMAIN, DOMAIN, config_entry.unique_id
+    )
+    assert entity_entry.unique_id != "light_1"
```

Comment:
you dont have an unique id, so remove that one.
The order is like:
1. add config entry ✅ 
2. `async_get_or_create` the entity
3. `async_setup` the entry. During startup it will trigger the migration
4. Check the entity registry and search for the entity. We should then see that the unique id is updated to be `f"{config_entry.entry_id}-1"`